### ISSUE TITLE
Danbooru Autotagger service integration doesn't work

### ISSUE THREAD

**Issue by Megum1n** [danbooru/danbooru#5223]
I couldn't find any other errors in the main services.

### Steps to reproduce
1. Download [docker-compose.yaml](https://github.com/danbooru/danbooru/blob/master/docker-compose.yaml)
2. Add `autotagger` service with `ghcr.io/danbooru/autotagger` image
3. Add `autotagger_url` = `http://autotagger:5000` into `danbooru_local_config.rb`
4. Upload an image from the frontend

### Expected result
1. Image is uploaded with auto generated tags

### Actual result
1. Error `Upload failed: AI tags is invalid.`
Result:
```json
{"status":"error","error":"AI tags is invalid","id":11,"source":null,"uploader_id":2,"created_at":"2022-07-01T19:14:01.539-04:00","updated_at":"2022-07-01T19:14:05.477-04:00","referer_url":null,"media_asset_count":1}
```

### Logs
Autotagger logs:
```json
{"time":"[01/Jul/2022:23:08:40 +0000]","id":"-","ip":"172.19.0.7","method":"POST","url":"/evaluate","status":"200","contentType":"-","userAgent":"Danbooru/a9fe73a4839a03ad9675ec54166e252d2bb5a4f9","referer":"-","sent":"1648","duration":"378819"}
```

Manual test:
```bash
curl http://danbooru.int.example.com:5000/evaluate -X POST -F file=@image.jpg -F format=json
```
```json
[{"filename":"wallhaven-72907.jpg","tags":{"1girl":0.9666173458099365,"rating:s":0.9450964331626892,"akemi_homura":0.9241950511932373,"long_hair":0.8888127207756042,"mahou_shoujo_madoka_magica":0.8792614340782166,"solo":0.8757835626602173,"black_hair":0.7547153830528259,"pantyhose":0.5542671084403992,"thighhighs":0.5110098123550415,"skirt":0.4701743423938751,"black_legwear":0.4015045464038849,"sitting":0.35237354040145874,"star_(sky)":0.2880863845348358,"night":0.26381242275238037,"sky":0.24046960473060608,"long_sleeves":0.22188153862953186,"hairband":0.19955317676067352,"ribbon":0.19881314039230347,"purple_eyes":0.1930823028087616,"boots":0.19133654236793518,"smile":0.18866387009620667,"zettai_ryouiki":0.18789975345134735,"looking_at_viewer":0.18276488780975342,"starry_sky":0.15281659364700317,"very_long_hair":0.14864397048950195,"original":0.14841994643211365,"bangs":0.13640281558036804,"holding":0.1297595202922821,"shirt":0.12569652497768402,"argyle_legwear":0.12441355735063553,"magical_girl":0.12303273379802704,"pleated_skirt":0.1225065290927887,"bow":0.12247934192419052,"hair_ribbon":0.11792214959859848,"outdoors":0.1113264188170433,"dress":0.10963766276836395,"space":0.10634855180978775,"gun":0.1044345274567604,"night_sky":0.10379345715045929,"gloves":0.10348358750343323,"twintails":0.10095828026533127}}]
```

**Comment by Megum1n** [comment]
The error is different on the latest beta builds.
`Upload failed: Failed to replace ai_tags because one or more of the new records could not be saved..`

**Comment by Megum1n** [comment]
```
E, [2022-07-03T07:50:27.097179 #501] ERROR -- : [c0f2ee03-4aeb-42d8-b577-0489c623f9ba] Module::DelegationError: MediaAsset#metadata delegated to media_metadata.metadata, but media_metadata is nil: #<MediaAsset id: 2901, created_at: "2022-07-03 03:48:14.267214000 -0400", updated_at: "2022-07-03 03:48:14.839663000 -0400", md5: "befe2a02e01a490321c92205a4f5fa7f", file_ext: "png", file_size: 6723127, image_width: 2934, image_height: 2057, duration: nil, status: "failed", file_key: "tvYhy12Dj", is_public: true>

def metadata(...);  _ = media_metadata;  _.metadata(...);rescue NoMethodError => e;  if _.nil? && e.name == :metadata;   raise DelegationError, "MediaAsset#metadata delegated to media_metadata.metadata, but media_metadata is nil: #{self.inspect}";  else;    raise;  end;end
                                                                                                                         ^^^^^
app/models/media_asset.rb:25:in `rescue in metadata'
app/models/media_asset.rb:25:in `metadata'
app/views/media_assets/show.html.erb:24
app/controllers/application_controller.rb:64:in `respond_with'
app/controllers/media_assets_controller.rb:18:in `show'
```

**Comment by Megum1n** [comment]
Rarely reproducible, but jfyi
I also got
```
Error:  undefined method `ai_tags' for nil:NilClass

      self.ai_tags = media_file.preview(360, 360).ai_tags # XXX should do this in parallel with thumbnail generation.
                                                 ^^^^^^^^
```

with empty `autotagger_url` parameter in the config.

**Comment by evazion** [comment]
Fixed in e44197a1ca969f21124254c1187b00823ea8d503 and 5c0055e2632f919205689b51b0d1f6a7ebb4e14e.
